cdap-proto/src/main/java/co/cask/cdap/proto/metadata/MetadataSearchTargetType.java

```diff
@@ -25,12 +25,12 @@
   // the custom values are required because these value match the entity-type stored as
   // a part of MDS key.
   ALL("All"),
-  ARTIFACT("Artifact"),
```

Comment:
let's chat about this today. I think this code even without your change depends on the class names of the ``Id.*`` classes, which seems bad. We may need to maintain a mapping between what gets serialized and what class to use to deserialize.